src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
 public class WorkWithFile {
+    private static final String SPLITTER = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final int NAME_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+    private StringBuilder textToWrite = new StringBuilder();
+
     public void getStatistic(String fromFileName, String toFileName) {
```

Comment:
It isn't good practice to make long methods, especially public ones.
Please split your solution to three private methods like readFile, generateReport and writeToFile.
Then in your public method make only calls to there 3 methods to make it short and clear